src/util/triplea/MapXMLCreator/TerritoryConnectionsPanel.java

```diff
@@ -0,0 +1,256 @@
+package util.triplea.MapXMLCreator;
+
+import games.strategy.util.AlphanumComparator;
+
+import java.awt.Color;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.Polygon;
+import java.awt.Shape;
+import java.awt.event.ActionEvent;
+import java.awt.event.MouseEvent;
+import java.awt.geom.Area;
+import java.awt.geom.Rectangle2D;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.swing.AbstractAction;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import util.image.ConnectionFinder;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class TerritoryConnectionsPanel extends ImageScrollPanePanel
+{
+	
+	String m_selectedTerritory = null;

```

Comment:
Please mark as private. Also look into the Java.util.Optional API. Please use Optionals instead of variables with null values.

This would then be transformed into:
`private Optional<String> selectedTerritory = Optional.empty()`